### PR TITLE
dNdS module for SMGs

### PR DESCRIPTION
**Pull request by Kdreval**
# Pull Request Checklists

**Important:** When opening a pull request, keep only the applicable checklist and delete all other sections.

### Required

- [X] I used the cookiecutter template and updated the placeholder rules.

- [X] The snakemake rules follow the [design guidelines](https://lcr-modules.readthedocs.io/en/latest/for_developers.html#module-rules). 

  - [X] All references to the `rules` object (_e.g._ for input files) are wrapped with `str()`.

- [X] Every rule in the module is either listed under `localrules` or has the `threads` and `resources` directives.

- [X] Input and output files are being symlinked into the `CFG["inputs"]` and `CFG["outputs"]` subdirectories, respectively.

- [ ] I grouped the input symlinking rule to the next job that uses the input files. 

- [X] I updated the final target rule (`*_all`) to include every output rule.

- [X] I explained important module design decisions in `CHANGELOG.md`.

- [X] I tested the module on real data for all supported `seq_type` values.

- [X] I updated the `default.yaml` configuration file to provide default values for each rule in the module snakefile.

- [X] I did not set any global wildcard constraints. Any/all wildcard constraints are set on a per-rule basis. 

- [X] I ensured that all symbolic links are relative and self-contained (_i.e._ do not point outside of the repository).

- [X] I replaced every value that should (or might need to) be updated in the default configuration file with `__UPDATE__`.

- [X] I recursively searched for all comments containing `TODO` to ensure none were left. For example:

  ```bash
  grep -r TODO modules/<module_name>/1.0
  ```

### CODE REVIEW

File: modules/dnds/1.0/config/default.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        options:
+            target_genes: "" # optionally, specify path to the genes of interest if this is a targeted sequencing
+                            # this will restrict SMG search to only these genes
+                            # this is a tsv file that contain one gene symbol/row and is expected to be without header
+            max_muts_per_gene_per_sample: Inf # How many mutations/gene dNdS should consider?
+                                            # This default value will consider all mutations/gene.
+                                            # When modified, this should be integer number.
+
+        conda_envs:
+            prepare_mafs: "{REPODIR}/envs/gatk/gatkR.yaml"
```
Comment (Jwong684) [inline modules/dnds/1.0/config/default.yaml:24]: is REPODIR a new variable now?

Comment (Kdreval) [reply modules/dnds/1.0/config/default.yaml:24]: Not exactly, it is just not used very often. This is a shortcut to the parent directory of lcr-modules. More details here: https://lcr-modules.readthedocs.io/en/latest/for_users.html?highlight=repodir#directory-placeholders

File: modules/dnds/1.0/src/R/run_dnds.R [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+# Load packages -----------------------------------------------------------
+message("Loading packages...")
+suppressWarnings(
+suppressPackageStartupMessages({
+    library(dndscv)
+})
+)
+
+# Read the maf file
+maf = read.table(
```
Comment (Jwong684) [inline modules/dnds/1.0/src/R/run_dnds.R:22]: would fread be better in this case?

**Review by Jwong684 — APPROVED** [review]
Seems good

File: modules/dnds/1.0/src/R/run_dnds.R (hunk 2 above)
Comment (Kdreval) [reply modules/dnds/1.0/src/R/run_dnds.R:22]: I opted for the base R alternative here to reduce the number of dependencies and make the conda environment more light. But in Rstudio will definitely opt for fread or read_tsv 😄

**Pull request merged**